File: androidutilslibrary/src/main/java/me/com/androidutilslibrary/DateUtils/DateUtils.java
package me.com.androidutilslibrary.DateUtils;

import android.net.ParseException;
import android.util.Log;

import java.text.SimpleDateFormat;
import java.util.Date;

public class DateUtils {
    /**
     * @param inputDate         string input Date
     * @param inputDatePattern  input date pattern
     * @param outputDatePattern output date pattern
     * @return for instance:
     * inputDatePattern: "yyyy-MM-dd", "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
     * outputDatePattern: "MMM, dd yyyy", "MMM dd, yyyy 'at' HH:mm a"
     */
    public static String getFormattedDateFromStringInputDate(String inputDate, String inputDatePattern, String outputDatePattern) {

        Date parsed;
        String outputDate = "";

        SimpleDateFormat df_input = new SimpleDateFormat(inputDatePattern, java.util.Locale.getDefault());
        SimpleDateFormat df_output = new SimpleDateFormat(outputDatePattern, java.util.Locale.getDefault());

        try {
            parsed = df_input.parse(inputDate);
            outputDate = df_output.format(parsed);

        } catch (ParseException e) {
            Log.d(DateUtils.class.getName(), e.getMessage());
        } catch (java.text.ParseException e) {
            e.printStackTrace();
        }

        return outputDate;
    }
}

File: androidutilslibrary/src/main/java/me/com/androidutilslibrary/dateutils/DateUtils.kt
package me.com.androidutilslibrary.dateutils

import android.net.ParseException
import android.util.Log
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {
    /**
     * @param inputDate         string input Date
     * @param inputDatePattern  input date pattern
     * @param outputDatePattern output date pattern
     * @return for instance:
     * inputDatePattern: "yyyy-MM-dd", "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
     * outputDatePattern: "MMM, dd yyyy", "MMM dd, yyyy 'at' HH:mm a"
     */
    fun getFormattedDateFromStringInputDate(inputDate: String, inputDatePattern: String, outputDatePattern: String): String {
        val parsed: Date?
        var outputDate = ""
        val dfInput = SimpleDateFormat(inputDatePattern, Locale.getDefault())
        val dfOutput = SimpleDateFormat(outputDatePattern, Locale.getDefault())
        try {
            parsed = dfInput.parse(inputDate)
            parsed?.let {
                outputDate = dfOutput.format(it)
            }
        } catch (e: ParseException) {
            Log.d(DateUtils::class.java.name, e.message ?: "")
        } catch (e: java.text.ParseException) {
            e.printStackTrace()
        }
        return outputDate
    }
}